### PR TITLE
Use parser plugin in from/read plugins

### PR DESCRIPTION
**Pull request by patszt**
This enables usage of a parser plugin whenever we encounter from X read FORMAT or read FORMAT from X in a pipeline.
Previously the FORMAT part was a NOP due to lack of a parser plugin.

### CODE REVIEW

**Review by jachris — APPROVED** [review]

File: libvast/builtins/operators/from_read.cpp [hunk 1]
```diff
@@ -167,13 +169,45 @@ class read_plugin final : public virtual operator_plugin {
...
+        caf::make_error(ec::lookup_error,
+                        fmt::format("no parser found for '{}'", parser_name)),
+      };
+    }
+    auto loader_name = std::string{};
+    auto loader_cfg = record{};
+    if (auto read_opt = std::get<1>(parsed)) {
+      loader_name = std::move(std::get<1>(*read_opt));
+    } else if (auto default_loader = parser->make_default_loader()) {
+      VAST_WARN("default loader");
```
[the remark this answers is not in the document]
Comment (patszt) [reply libvast/builtins/operators/from_read.cpp:186]: definitely!

File: libvast/builtins/operators/from_read.cpp [hunk 2]
```diff
@@ -33,34 +33,36 @@ class from_operator final : public crtp_operator<from_operator> {
...
+  operator()(generator<chunk_ptr> input, operator_control_plane& ctrl) const
     -> caf::expected<generator<table_slice>> {
-    // TODO: Implement this.
-    return caf::make_error(ec::unimplemented,
-                           "parser_plugin does not exist yet");
+    auto parser = parser_plugin_.make_parser(config_, ctrl);
+    if (not parser) {
+      return std::move(parser.error());
+    }
+    return (*parser)(std::move(input));
```
[the remark this answers is not in the document]
Comment (patszt) [reply libvast/builtins/operators/from_read.cpp]: Completed in https://github.com/tenzir/vast/pull/3072/commits/0711fb427fe5648363f138c97ea96a8e8d4840cf

File: libvast/builtins/operators/from_read.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (patszt) [reply libvast/builtins/operators/from_read.cpp]: Completed in https://github.com/tenzir/vast/pull/3072/commits/06e67fcc24dd2957bb4ea19397c24a2faddb3f65.

File: libvast/test/loader_plugin.cpp [hunk 3]
```diff
@@ -162,29 +141,41 @@ TEST(stdin loader - one complete chunk) {
 }
 
 TEST(stdin loader - from operator) {
```
[the remark this answers is not in the document]
Comment (patszt) [reply libvast/test/loader_plugin.cpp]: https://github.com/tenzir/vast/pull/3072/commits/06e67fcc24dd2957bb4ea19397c24a2faddb3f65

**Pull request merged**